app/dao/notifications_dao.py

```diff
@@ -136,6 +135,28 @@ def dao_get_7_day_agg_notification_statistics_for_service(service_id,
 
 
 @statsd(namespace="dao")
+def dao_get_template_usage(service_id, limit_days=None):
+
+    query = db.session.query(
+        func.count(NotificationHistory.template_id).label('count'),
+        NotificationHistory.template_id,
+        func.DATE(NotificationHistory.created_at).label('day'),
+        Template.name,
+        Template.template_type
+    )
+
+    query_filter = [NotificationHistory.service_id == service_id]
+    if limit_days is not None:
+        query_filter.append(NotificationHistory.created_at >= days_ago(limit_days))
+
+    return query.filter(*query_filter) \
+        .join(Template)\
```

Comment:
This should really be joining to TemplateHistory - the query results shouldn't change, but it means that in future we can remove the fkey relationship completely (but only after 
https://github.com/alphagov/notifications-api/pull/599 is merged)